tests/components/LocationBias.test.tsx

```diff
@@ -0,0 +1,143 @@
+  location: {
+    locationBias: {
+      latitude: 40.741591687843005,
+      longitude: -74.00530254443494,
+      displayName: 'New York City, NY',
+      method: LocationBiasMethod.Ip
+    }
+  }
+};
+
+const mockedStateNoDisplayName: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: null,
+      longitude: null,
+      displayName: null,
+      method: null
+    }
+  },
+};
+
+const newGeoPosition = {
+  coords: {
+    accuracy: null,
+    altitude: null,
+    altitudeAccuracy: null,
+    heading: null,
+    latitude: 40.741591687843005,
+    longitude: -74.00530254443494,
+    speed: null,
+  },
+  timestamp: null
+};
+
+beforeEach(() => {
+  mockAnswersHooks({
+    mockedState: mockedStateVaDevice,
+    mockedActions: {
+      state: mockedStateVaDevice,
+      setUserLocation: jest.fn(),
+    }
+  });
+  jest.spyOn(locationOperations, 'getUserLocation').mockResolvedValue(newGeoPosition);
+  jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+});
+
+it('renders the proper text (location DisplayName, method, and update btn)', () => {
+  render(<LocationBias />);
+  const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+  const locationNameElement = screen.getByText(expectedLocationName);
+  expect(locationNameElement).toBeDefined();
+
+  const expectedMethodMessage = '(based on your device) -';
+  const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+  expect(basedOnMethodElement).toBeDefined();
+
+  const updateLocationButton = screen.getByText('Update your location');
+  expect(updateLocationButton).toBeDefined();
+});
+
+it('calls setUserLocation with coordinates returned by getUserLocation as params when update location clicked', async () => {
+  const actions = spyOnActions();
+  render(<LocationBias />);
+  clickUpdateLocation();
+
+  const expectedCoordinates = {
+    latitude: newGeoPosition.coords.latitude,
+    longitude: newGeoPosition.coords.longitude
+  };
+
+  expect(locationOperations.getUserLocation).toBeCalled();
+  await waitFor(() => {
+    expect(actions.setUserLocation).toBeCalledWith(expectedCoordinates);
+  });
+});
+
+it('updates rendered DisplayName if location changes and update button is clicked', async () => {
+  render(<LocationBias />);
+  const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+  const locationNameElement = screen.getByText(expectedLocationName);
+  expect(locationNameElement).toBeDefined();
+
+  mockAnswersState(mockedStateNyIP);
+  clickUpdateLocation();
+
+  await waitFor(() => {
+    expect(searchOperations.executeSearch).toBeCalled();
+  });
+
+  const newExpectedLocationName = mockedStateNyIP.location.locationBias.displayName;
+  const newLocationNameElement = screen.getByText(newExpectedLocationName);
+  expect(newLocationNameElement).toBeDefined();
+});
+
+it('renders correct attribution message, device', () => {
+  mockAnswersState(mockedStateVaDevice);
```

Comment:
is this mock statement needed? I thought by default it would have the VA location in state?

Reply:
No, I don't think it is needed--  when I was deciding on whether or not to include that line, I was considering that it might be more clear when reading it later since there's also a test for IP that requires a change from default.  Good to know that I should not do that, though!